avm/res/web/hosting-environment/main.bicep

```diff
@@ -229,11 +210,12 @@ resource appServiceEnvironment_roleAssignments 'Microsoft.Authorization/roleAssi
   for (roleAssignment, index) in (roleAssignments ?? []): {
     name: guid(appServiceEnvironment.id, roleAssignment.principalId, roleAssignment.roleDefinitionIdOrName)
     properties: {
-      roleDefinitionId: contains(builtInRoleNames, roleAssignment.roleDefinitionIdOrName)
-        ? builtInRoleNames[roleAssignment.roleDefinitionIdOrName]
-        : contains(roleAssignment.roleDefinitionIdOrName, '/providers/Microsoft.Authorization/roleDefinitions/')
-            ? roleAssignment.roleDefinitionIdOrName
-            : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName)
+      roleDefinitionId: builtInRoleNames[?roleAssignment.roleDefinitionIdOrName] ?? (contains(
```

Comment:
Should be updated to latest [spec](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#role-assignments). Please also not the additional variable `formattedRoleAssignments`.
You can see an example implementation here: https://github.com/Azure/bicep-registry-modules/pull/3520/files

Note that this will also close one of the line items here: #2973

Reply:
Hey @AlexanderSehr  - Changes are in. Let us know if I have missed anything?

Reply:
Looks great. @tsc-buddy. Please merge at will (and keep an eye on the workflow to ensure the version is published 😉)